src/main/java/com/lesson52/service/StudentService.java

```diff
@@ -0,0 +1,127 @@
+
+    @Autowired
+    private StudentRepository studentRepository;
+
+    @Autowired
+    private GroupRepository groupRepository;
+
+
+    public List<StudentDTO> getStudentsByGroup(String title) {
+        List<Student> students = studentRepository.getStudentsByGroupIgnoreCaseTitle(title);
+        List<StudentDTO> studentDTOList = students.stream()
+                .map(StudentMapper::toDTO)
+                .collect(Collectors.toList());
+        return studentDTOList;
+    }
+
+    public List<StudentDTO> getUnpaidStudents() {
+        List<Student> studentList = studentRepository.findStudentByUnPayIsFalse();
+
+        List<StudentDTO> studentDTOList = studentList.stream()
+                .map(StudentMapper::toDTO)
+                .collect(Collectors.toList());
+
+        return studentDTOList;
+    }
+
+    public List<StudentDTO> getStudentByName(String name) {
+        List<Student> studentList = studentRepository.findByNameIgnoreCase(name);
+
+        List<StudentDTO> studentDTOList = studentList.stream()
+                .map(StudentMapper::toDTO)
+                .collect(Collectors.toList());
+
+        return studentDTOList;
+    }
+
+    public List<StudentDTO> findStudentBySurname(String surname) {
+        List<Student> studentList = studentRepository.findStudentBySurnameIgnoreCase(surname);
+
+        if (studentList.isEmpty()) {
+            return Collections.emptyList();
+        }
+        List<StudentDTO> studentDTOList = studentList.stream()
+                .map(StudentMapper::toDTO)
+                .collect(Collectors.toList());
+
+        return studentDTOList;
+    }
+
+    public Student updateStudent(StudentDetailsDTO studentDetailsDTO) {
+
+        Optional<Student> optionalStudent = studentRepository.findById(studentDetailsDTO.getId());
+
+        if (optionalStudent.isPresent()) {
+            Student student = optionalStudent.get();
+
+            student.setName(studentDetailsDTO.getName());
+            student.setSurname(studentDetailsDTO.getSurname());
+            student.setUnPay(studentDetailsDTO.isUnPay());
+
+            studentRepository.save(student);
+            return student;
+        } else {
+            return null;
+        }
+    }
+
+    public boolean deleteStudent(int id) {
+        boolean isExistsStudent = studentRepository.existsById(id);
+        if (isExistsStudent) {
+            studentRepository.deleteById(id);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public Student createStudent(StudentDTO studentDTO) {
+        Student entityStudent = StudentMapper.toEntity(studentDTO);
+
+        if (entityStudent.getGroup() != null) {
+            Group group = groupRepository.findById(entityStudent.getGroup().getId()).orElse(null);
+            if (group != null) {
+                System.out.println("GROUP " +group);
+                entityStudent.setGroup(group);
+            }else {
+                return null;
```

Comment:
It's not correct to return null.
If you want to tell about exception situaion, you should provide an exception here